build/preview.ts

```diff
@@ -0,0 +1,40 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+async function run() {
+  const cwd = path.resolve(__dirname);
+
+  const buildDir = path.basename(cwd);
+  const repoDir = cwd.replace(buildDir, '');
+
+  // Patch README
+  const data = fs.readFileSync(path.join(repoDir, 'README.md'), { encoding: 'utf8' });
+  const insert = fs.readFileSync(path.join(buildDir, 'nightly.md'), { encoding: 'utf8' });
+  fs.writeFileSync(path.join(repoDir, 'README.md'), `${insert}\n${data}`);
+
+  // Patch package.json
+  const date = new Date(new Date().toLocaleString('en-US', { timeZone: 'America/New_York' }));
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  let json = require('../package.json');
+
+  const version = `${String(date.getFullYear())}.${date.getMonth() + 1}.${date.getDate()}${String(
+    date.getHours(),
+  ).padStart(2, '0')}`;
+
+  json = JSON.stringify(
+    {
+      ...json,
+      preview: true,
+      // name: `${json.name}-preview`,
```

Comment:
I think we need the `-preview` suffix to the name?

Reply:
During publish testing I couldn't have that there. Nice catch!

Reply:
With the new matrix approach we no longer need this file